website/client/components/appMenu.vue

```diff
@@ -353,13 +373,16 @@ export default {
     openPartyModal () {
       this.$root.$emit('show::modal', 'create-party-modal');
     },
-    showBuyGemsModal () {
+    showBuyGemsModal (startingPage) {
+      this.$store.state.gemModalOptions.startingPage = startingPage;
+
       Analytics.track({
         hitType: 'event',
         eventCategory: 'button',
         eventAction: 'click',
         eventLabel: 'Gems > Toolbar',
       });
+
       this.$root.$emit('show::modal', 'buy-gems');
```

Comment:
not completely related to this PR but this $emit should be `this.$root.$emit('show::modal', 'buy-gems', {alreadyTracked: true})` or it'll be tracked twice, see https://github.com/HabitRPG/habitica/blob/develop/website/client/app.vue#L220 (I had to add that because not all modals are shown with `this.$root.$emit('show::modal', 'buy-gems');` so I cannot track them like here)